### DISCUSSION TITLE
Programmatic or OPENAPI or any other API based method to read the contents of Report v3 columns and reports

### DISCUSSION THREAD

**Discussion by gourav-sg** [amzn/ads-advanced-tools-docs#253] [Q&A]
Dear friends,

may be this is a novice question, but I was wondering if there is an API interface to read the contents that are hosted here:

* https://advertising.amazon.com/API/docs/en-us/guides/reporting/v3/columns
* https://advertising.amazon.com/API/docs/en-us/guides/reporting/v3/report-types/

Regards,
Gourav Sengupta

**Comment by elizafarley** [comment]
Hi @gourav-sg, currently there isn't any programmatic interface to get this info. You need to rely on the documentation you referenced above. 

Can you say more about how a programmatic interface would be helpful for you? If you think it would be a useful feature, I'm happy to pass that feedback onto the product team.

**Comment by gourav-sg** [comment]
Dear friend,

Surely

Manually typing the fields and their data type is a tedious task and
painful. If exposed as a file then it becomes easier to do.

So we can programmatically verify field name changes across versions, data
types, etc.

Regards,
Gourav Sengupta

**Comment by gourav-sg** [comment]
Hi,
is there any update on this simple request? Having APIs programmatically list out the schemas is a good practice and ensures that users of the API solutions get more stable and reliable interface. Copying and pasting from static documents perhaps is for reading and exploring.

Regards,
Gourav
